src/mlpack/tests/sparse_autoencoder_test_2.cpp

```diff
@@ -0,0 +1,278 @@
+/**
+ * @file sparse_autoencoder_test.cpp
+ * @author Siddharth Agrawal
+ *
+ * Test the SparseAutoencoder class.
+ */
+
+#include <mlpack/methods/ann/sparse_autoencoder_function.hpp>
+#include <mlpack/methods/ann/activation_functions/lazy_logistic_function.hpp>
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/sparse_autoencoder/sparse_autoencoder.hpp>
+#include <mlpack/methods/ann/layer/base_layer.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "old_boost_test_definitions.hpp"
+
+using namespace mlpack;
+using namespace arma;
+
+using FSigmoidLayer = ann::SigmoidLayer<ann::LazyLogisticFunction>;
+
+//sparse autoencoder function
+using SAEF = ann::SparseAutoencoderFunction<FSigmoidLayer, FSigmoidLayer>;
+//sparse autoencoder function greedy
+using SAEFG = ann::SparseAutoencoderFunction<FSigmoidLayer, FSigmoidLayer, std::true_type>;
+
+BOOST_AUTO_TEST_SUITE(SparseAutoencoderTest2);
```

Comment:
Just a note -- when this is merged, we can remove the old `sparse_autoencoder_test.cpp` and revert this to `SparseAutoencoderTest` instead of `SparseAutoencoderTest2`.